### DISCUSSION TITLE
v0.8.0

### DISCUSSION THREAD

**Discussion by chiamp** [google/flax#3648] [General]
## What's Changed
* bump version number by @levskaya in https://github.com/google/flax/pull/3446
* Add merge / finalize step when using OCDBT driver. Files will be first written to per-process subdirectories, which are later copied by reference to the main directory before the checkpoint is finalized. by @copybara-service in https://github.com/google/flax/pull/3426
* fixed quickstart by @chiamp in https://github.com/google/flax/pull/3451
* [NVIDIA] Update the algorithm to compute fp8 scales by @kaixih in https://github.com/google/flax/pull/3441
* added pre-commit hook that sort imports and formats by @chiamp in https://github.com/google/flax/pull/3455
* restructured doc folders by @chiamp in https://github.com/google/flax/pull/3434
* Forked a subset of JAX configuration APIs by @superbobry in https://github.com/google/flax/pull/3448
* Fix Module.clone in deepclone mode for internal usage. by @levskaya in https://github.com/google/flax/pull/3459
* Add user-friendly module copy method. by @levskaya in https://github.com/google/flax/pull/3461
* Add simple argument-only lifted nn.grad function. by @levskaya in https://github.com/google/flax/pull/3463
* exempt a jax.config deprecation warning by @levskaya in https://github.com/google/flax/pull/3465
* Clean up pyproject.toml. by @levskaya in https://github.com/google/flax/pull/3468
* Allow for fast accumulation selection for FP8 GEMM by @wenscarl in https://github.com/google/flax/pull/3416
* re-added quickstart guide by @chiamp in https://github.com/google/flax/pull/3471
* fixed tabulate docstring by @chiamp in https://github.com/google/flax/pull/3452
* Add NNX by @cgarciae in https://github.com/google/flax/pull/3218
* Bump pillow from 9.5.0 to 10.0.1 in /examples/vae by @dependabot in https://github.com/google/flax/pull/3390
* updated attention_test by @chiamp in https://github.com/google/flax/pull/3454
* [nnx] Improve docs by @cgarciae in https://github.com/google/flax/pull/3478
* added example docstrings by @chiamp in https://github.com/google/flax/pull/3453
* fix nn.value_and_grad by implementing directly in core by @levskaya in https://github.com/google/flax/pull/3479
* Add dataset loading guide (Issue #2116) by @VictorPrins in https://github.com/google/flax/pull/3450
* [nnx] Add support for python container types by @cgarciae in https://github.com/google/flax/pull/3486
* remove SelfAttention test and warning filter by @chiamp in https://github.com/google/flax/pull/3470
* disabled ruff formatter by @chiamp in https://github.com/google/flax/pull/3482
* adding doctest to .rst files by @chiamp in https://github.com/google/flax/pull/3481
* changed pip installs to use quotes by @chiamp in https://github.com/google/flax/pull/3477
* added enum support for tabulate by @chiamp in https://github.com/google/flax/pull/3485
* fix bug in optimizer-api.md by @zhaoyang-0204 in https://github.com/google/flax/pull/3462
* removed selfattention from doctest by @chiamp in https://github.com/google/flax/pull/3489
* [nnx] Add missing import on why.ipynb by @cgarciae in https://github.com/google/flax/pull/3503
* [nnx] switch to nested State representation by @cgarciae in https://github.com/google/flax/pull/3502
* Improved Rigor of `PReLU` Test by @Micky774 in https://github.com/google/flax/pull/3498
* added geglu activation and tests by @HMUNACHI in https://github.com/google/flax/pull/3512
* [nnx] Add LinearGeneral and MultiHeadAttention by @cgarciae in https://github.com/google/flax/pull/3487
* Add NNX/Linen consistency test for `Embed` layer by @Micky774 in https://github.com/google/flax/pull/3513
* Add NNX/Linen API consistency test for `Conv` layer by @Micky774 in https://github.com/google/flax/pull/3511
* Prevent crash in dataclasses with no-init params by @NeilGirdhar in https://github.com/google/flax/pull/3514
* [nnx] Variable referece sharing by @cgarciae in https://github.com/google/flax/pull/3516
* Added NNX/Linen API consistency test for `Linear/Dense` layer by @Micky774 in https://github.com/google/flax/pull/3509
* Add missing mask argument to LayerNorm, RMSNorm, and GroupNorm. by @carlosgmartin in https://github.com/google/flax/pull/3510
* [nnx] Fix graph_utils bug by @cgarciae in https://github.com/google/flax/pull/3518
* remove deprecated normalize function by @chiamp in https://github.com/google/flax/pull/3531
* Reduced number of parameterizations for `Conv` NNX/Linen consistency test by @Micky774 in https://github.com/google/flax/pull/3526
* Ensure that `_hashable_filter` does not convert strings to a tuple of letters by @copybara-service in https://github.com/google/flax/pull/3533
* added sow attention weights by @chiamp in https://github.com/google/flax/pull/3529
* Fix scan out_axes by @cgarciae in https://github.com/google/flax/pull/3540
* updated embed docstring by @chiamp in https://github.com/google/flax/pull/3539
* add test_scan_negative_axes by @cgarciae in https://github.com/google/flax/pull/3542
* add module methods to api docs by @chiamp in https://github.com/google/flax/pull/3544
* fixed double backquote code font by @chiamp in https://github.com/google/flax/pull/3545
* add nnx conv support for int kernel size by @chiamp in https://github.com/google/flax/pull/3537
* added sow attention weights to NNX by @chiamp in https://github.com/google/flax/pull/3548
* changed `return_weights` to `sow_weights` for attention layer by @chiamp in https://github.com/google/flax/pull/3550
* format linen_linear_test.py by @chiamp in https://github.com/google/flax/pull/3553
* re-factored features arg by @chiamp in https://github.com/google/flax/pull/3554
* updated NNX readme by @chiamp in https://github.com/google/flax/pull/3556
* Disable ruff sort imports by @cgarciae in https://github.com/google/flax/pull/3560
* Add StateVariablesMapping by @cgarciae in https://github.com/google/flax/pull/3523
* add kwargs support for nn.jit by @copybara-service in https://github.com/google/flax/pull/3559
* [nnx] Fix readme install instruction by @cgarciae in https://github.com/google/flax/pull/3565
* implement Rng.__getattr__ by @cgarciae in https://github.com/google/flax/pull/3547
* [nnx] add qkv_features back to MHA by @cgarciae in https://github.com/google/flax/pull/3566
* updated readme by @chiamp in https://github.com/google/flax/pull/3563
* fixed typo by @chiamp in https://github.com/google/flax/pull/3561
* Raise an error for a bad key type by @NeilGirdhar in https://github.com/google/flax/pull/3527
* re-factored nnx initializers by @chiamp in https://github.com/google/flax/pull/3555
* [nnx] Add complex test with scan + batchnorm + dropout by @cgarciae in https://github.com/google/flax/pull/3567
* [nnx] Add interacting with JAX section to README by @cgarciae in https://github.com/google/flax/pull/3573
* expose ones and zeros initializers by @chiamp in https://github.com/google/flax/pull/3574
* Fix promotion bug in MultiHeadDotProductAttention: by @giovannic in https://github.com/google/flax/pull/3571
* fixed error doc formatting by @chiamp in https://github.com/google/flax/pull/3587
* [nnx] Improve spmd by @cgarciae in https://github.com/google/flax/pull/3580
* [nnx] improve graph_utils._set_key_tuple by @cgarciae in https://github.com/google/flax/pull/3592
* [nnx] Fix variable unflatten by @cgarciae in https://github.com/google/flax/pull/3578
* [nnx] add open in colab button to why nnx by @cgarciae in https://github.com/google/flax/pull/3596
* [nnx] Export missing symbols by @cgarciae in https://github.com/google/flax/pull/3583
* [nnx] flaglib add get overloads by @cgarciae in https://github.com/google/flax/pull/3582
* Fix type in NNX readme by @shoyer in https://github.com/google/flax/pull/3591
* [nnx] add submodule iterator by @cgarciae in https://github.com/google/flax/pull/3581
* [nnx] delete flaglib duplicated copyright comment by @cgarciae in https://github.com/google/flax/pull/3600
* fixed NNX decode and dynamic slicing by @chiamp in https://github.com/google/flax/pull/3576
* [nnx] cleanup CallableProxy by @cgarciae in https://github.com/google/flax/pull/3608
* [nnx] improve runtime flags by @cgarciae in https://github.com/google/flax/pull/3607
* fixed broken links on quick start guide by @chiamp in https://github.com/google/flax/pull/3610
* added multiheadattention alias by @chiamp in https://github.com/google/flax/pull/3572
* Rollback of Copybara import of the project: by @copybara-service in https://github.com/google/flax/pull/3612
* add missing docs for module functions by @cgarciae in https://github.com/google/flax/pull/3619
* fix lm1b data sharding by @cgarciae in https://github.com/google/flax/pull/3620
* improve embed by @jianyizh in https://github.com/google/flax/pull/3590
* disable ruff linter by @chiamp in https://github.com/google/flax/pull/3625
* Add compact_name_scope decorator by @cgarciae in https://github.com/google/flax/pull/3621
* Copybara import of the project: by @copybara-service in https://github.com/google/flax/pull/3638
* added BatchApply by @chiamp in https://github.com/google/flax/pull/3634
* add compact_name_scope v2 by @copybara-service in https://github.com/google/flax/pull/3640
* add compact_name_scope v2 by @copybara-service in https://github.com/google/flax/pull/3642
* release 0.8.0 by @chiamp in https://github.com/google/flax/pull/3644

## New Contributors
* @superbobry made their first contribution in https://github.com/google/flax/pull/3448
* @VictorPrins made their first contribution in https://github.com/google/flax/pull/3450
* @zhaoyang-0204 made their first contribution in https://github.com/google/flax/pull/3462
* @Micky774 made their first contribution in https://github.com/google/flax/pull/3498
* @HMUNACHI made their first contribution in https://github.com/google/flax/pull/3512
* @carlosgmartin made their first contribution in https://github.com/google/flax/pull/3510
* @giovannic made their first contribution in https://github.com/google/flax/pull/3571
* @jianyizh made their first contribution in https://github.com/google/flax/pull/3590

**Full Changelog**: https://github.com/google/flax/compare/v0.7.5...v0.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/google/flax/releases/tag/v0.8.0'>v0.8.0</a>.</em>
